tcms/testplans/urls/plan_urls.py

```diff
@@ -5,9 +5,10 @@
 from tcms.testplans import views
 
 urlpatterns = [
-    url(r'^(?P<plan_id>\d+)/$', views.get, name='test_plan_url_short'),
+    url(r'^(?P<plan_id>\d+)/$', views.TestPlanGetView.as_view(), name='test_plan_url_short'),
     # human friendly url
-    url(r'^(?P<plan_id>\d+)/(?P<slug>[-\w\d]+)$', views.get, name='test_plan_url'),
+    url(r'^(?P<plan_id>\d+)/(?P<slug>[-\w\d]+)$', views.TestPlanGetView.as_view(),
+        name='test_plan_url'),
```

Comment:
This is existing functionality, which is best to be implemented in 2 separate views: 

1) If accessing the plan only via ID -> redirect to the full URL
2) Show the plan if visited via full URL